### PR TITLE
Some corrections and additions

### PR DESCRIPTION
**Pull request by slang25**
A few suggestions

### CODE REVIEW

**Comment by AnthonySteele** [comment]
Does the ThreadPoool really not have a sync context at all? I based that on your table and on https://msdn.microsoft.com/en-us/magazine/gg598924.aspx 
If the 2 columns of the table are identical , I might as well collapse them to one.

File: AsyncResync.md [hunk 1]
```diff
@@ -75,9 +75,11 @@ finally
 
 This has the advantage over `Task.Run` of not using an additional thread for the initial synchronous invocations, and applying this technique at multiple levels comes at no extra cost, whereas each `Task.Run` with cost yet another thread.
 
-### Use a library that has a custom synchronisation context.
+### Run your code in in a single threaded fasion using `JoinableTaskFactory`
```
Comment (AnthonySteele) [inline AsyncResync.md]: "fashion"

File: AsyncResync.md [hunk 2]
```diff
@@ -75,9 +75,11 @@ finally
...
 
-You aren't going to want to write your own synchronisation context, but you can  e.g. use a library like [vs-threading](https://github.com/Microsoft/vs-threading/)
+Rather than just mitigating from deadlocks, you can run your code using a `JoinableTaskFactory` from [vs-threading](https://github.com/Microsoft/vs-threading/). `JoinableTaskFactory` internally sets a custom `SynchronizationContext` that will pass the synchronous continuations back to the blocking thread that was used to initially wait with. This results in a single threaded execution, which is more economical with your precious threads.
```
Comment (AnthonySteele) [inline AsyncResync.md]: Is this the only library worth recommending?

**Comment by slang25** [comment]
Regarding the thread pool, if I queue some work on the thread pool, then `SynchronizationContext.Current` is null, just like on the entry thread of a console application. When you are responding to a web request in ASP.NET, you are on a thread pool thread (you can check with [IsThreadPoolThread](https://msdn.microsoft.com/en-us/library/system.threading.thread.isthreadpoolthread(v=vs.110).aspx)), the difference is that ASP.NET has first called `SynchronizationContext.SetSynchronizationContext(new AspNetSynchronizationContext())`.

The thing that's a little weird and confusing is that the table refers to "Default", what I understand that to mean is `SynchronizationContext`, which is the base class for others, and the implementation does behave like the table says, but when you do `await`, there are explicit checks in the framework that say: `if(capturedContext == null || capturedContext.GetType() == typeof(SynchronizationContext)) // then use the current TaskScheduler`
So if you do `SynchronizationContext.SetSynchronizationContext(new SynchronizationContext())`, you aren't actually using it when you await. [See here for more](https://stackoverflow.com/a/20956722/2225808)

**Comment by AnthonySteele** [comment]
Every time that i think that I know it, it turns out that it's actually more complicated :(

**Pull request merged**